source/js/components/inputItem.js

```diff
@@ -1,122 +1,102 @@
-									onChange={this.onInputChange} 
-									onFocus={this.clearValidityMessage} /> <label htmlFor="isRequired">Required</label>
-							</div>
-						</div>
-					</form>
-			</li>
-		);
-	}
+const onInputChange = (e, {id, editInputItem}) => {
+	const target = e.target;
+	const name = target.name;
+	const value = target.type === 'checkbox' ? target.checked : target.value;
+	editInputItem({
+		id,
+		[name]: value
+	});
+};
 
-}
+const setValidityMessage = ({id, editInputItem}, isValid) => {
+	editInputItem({
+		id,
+		isValid: isValid
+	});
+};
 
-InputItem.defaultProps = {
-  isRequired: false
+const renderInputString = (props) => {
+	const typeStr = ` type="${props.inputType}"`;
+	const patternStr = props.inputPattern ?
+		` pattern="${props.inputPattern}"` :
+		'';
+	const requiredStr = props.isRequired ?
+		` required` :
+		'';
+	return `input${typeStr}${patternStr}${requiredStr}`;
 };
 
-export default InputItem;
\ No newline at end of file
+const InputItem = (props) => (
+	<li className="input-item" id={props.id} style={props.style}>
+		<form onSubmit={(e) =>onFormSubmit(e, props)}>
+			<button className="input-item__remove-button" type="button" onClick={() => props.removeInputItem(props.id)}>&times;</button>
+			<div className="input-item__display">
+				<span className="input-item__validity-display">{props.isValid === true ? 'Form submitted! No validation errors.' : ''}</span>
+				<input 
+					className="input-item__display-input"
+					type={props.inputType} 
+					pattern={props.inputPattern} 
+					required={props.isRequired} 
+					onFocus={() => setValidityMessage(props, false)} />
+				<pre>&lt;{renderInputString(props)}&gt;</pre>
+				<button className={submitButtonClasses} type="submit">Test This Input</button>
+			</div>
+			<div className="input-item__controls">
+				<div className="input-item__control">
+					<label htmlFor="inputType">Input Type:</label>
+					<select 
+						name="inputType"
+						id="inputType"
+						value={props.inputType} 
+						onChange={(e) => onInputChange(e, props)} 
+						onFocus={() => setValidityMessage(props, false)}>
+						{types.map(type => <option key={type.value} value={type.value}>{type.name}</option>)}
+					</select>
+				</div>
+				<div className="input-item__control">
+					<label htmlFor="inputPattern">Pattern:</label>
+					<select 
+						name="inputPattern"
+						id="inputPattern"
+						value={props.inputPattern} 
+						onChange={(e) => onInputChange(e, props)}
+						onFocus={() => setValidityMessage(props, false)}>
+						{patterns.map(pattern => <option key={pattern.value} value={pattern.value}>{pattern.name}</option>)}
+					</select>
+				</div>
+				<div className={requiredClasses}>
+					<input 
+						name="isRequired"
+						id="isRequired"
+						type="checkbox" 
+						checked={props.isRequired} 
+						onChange={(e) => onInputChange(e, props)}
+						onFocus={() => setValidityMessage(props, false)} /> <label htmlFor="isRequired">Required</label>
```

Comment:
Not sure what the purpose of this onFocus event is for the required checkbox.

Reply:
If you submit a form with no errors, there's a message that displays. I have an onFocus on every input to clear that message whenever you re-interact with the form.